numba/types/functions.py

```diff
@@ -87,7 +84,7 @@ def __init__(self, template):
             keys = set(temp.key for temp in self.templates)
             if len(keys) != 1:
                 raise ValueError("incompatible templates: keys = %s"
-                                 % (this,))
+                                 % (keys,))
```

Comment:
This snippet may be useful to check that this is now working as expected:
```python
from numba.typing.templates import make_overload_template
from numba import types

def func_stub():
    pass

def func_stub2():
    pass

def ol():
    pass

template1 = make_overload_template(func_stub, ol, {}, True)
template2 = make_overload_template(func_stub2, ol, {}, True)

types.Function((template1, template2)) # should fail
```

Reply:
I can confirm the above snippet gives

```
NameError: name 'this' is not defined
```

on `master`, and 

```
ValueError: incompatible templates: keys = {<function func_stub2 at 0x10d236158>, <function func_stub at 0x10ce720d0>}
```

with this PR

Reply:
Thanks for confirming @jrbourbeau